src/xpk/core/kjob.py

```diff
@@ -0,0 +1,196 @@
+
+app_profile_yaml = """
+apiVersion: kjobctl.x-k8s.io/v1alpha1
+kind: ApplicationProfile
+metadata:
+  name: {name}
+  namespace: default
+spec:
+  supportedModes:
+    - name: Slurm
+      template: {template}
+      requiredFlags: []
+"""
+
+
+def verify_kjob_installed(args: Namespace) -> None:
+  """Check if kjob is installed. If not provide user with proper communicate and exit.
+  Args:
+    args - user provided arguments.
+  Returns:
+    None
+  """
+  command = "kubectl-kjob help"
+  task = "Verify kjob installation "
+  verify_kjob_installed_code, _ = run_command_for_value(command, task, args)
+
+  if verify_kjob_installed_code == 0:
+    xpk_print("kjob found")
+
+  if verify_kjob_installed_code != 0:
+    xpk_print(
+        " kjob not found. Please follow"
+        " https://github.com/kubernetes-sigs/kueue/blob/main/cmd/experimental/kjobctl/docs/installation.md"
+        " to install kjob."
+    )
+    xpk_exit(verify_kjob_installed_code)
+
+
+def create_app_profile_instance(args: Namespace) -> None:
+  """Create new AppProfile instance on cluster with default settings.
+
+  Args:
+    args - user provided arguments
+  Returns:
+    None
+  """
+  yml_string = app_profile_yaml.format(
+      name=APP_PROFILE_TEMPLATE_DEFAULT_NAME,
+      template=JOB_TEMPLATE_DEFAULT_NAME,
+  )
+
+  tmp = write_tmp_file(yml_string)
+  command = f"kubectl apply -f {str(tmp.file.name)}"
+  return_code = run_command_with_updates(command, "Creating AppProfile", args)
+  if return_code != 0:
+    xpk_exit(return_code)
+
+
+def create_job_template_instance(args: Namespace) -> None:
+  """Create new JobTemplate instance on cluster with default settings.
+
+  Args:
+    args - user provided arguments
+  Returns:
+    None
+  """
+  yml_string = job_template_yaml.format(
+      name=JOB_TEMPLATE_DEFAULT_NAME,
+      parallelism=JOB_TEMPLATE_DEFAULT_PARALLELISM,
+      completions=JOB_TEMPLATE_DEFAULT_COMPLETIONS,
+      container=JOB_TEMPLATE_DEFAULT_CONT_NAME,
+      image=JOB_TEMPLATE_DEFAULT_IMG,
+  )
+
+  tmp = write_tmp_file(yml_string)
+  command = f"kubectl apply -f {str(tmp.file.name)}"
+  return_code = run_command_with_updates(command, "Creating JobTemplate", args)
+  if return_code != 0:
+    xpk_exit(return_code)
+
+
+def download_files_from_github_into_dir(
+    path: str, urls: list[(str, str)]
+) -> None:
+  for url, fn in urls:
+    target = join(path, fn)
+    try:
+      urllib.request.urlretrieve(url, target)
+    except ContentTooShortError as e:
+      xpk_print(f"downloading kjob CRD {fn} failed due to {e.content}")
+      xpk_exit(1)
+
+
+def apply_kjob_crds(args: Namespace) -> None:
+  """Apply kjob CRDs on cluster.
+
+  This function downloads kjob CRDs files from kjob repo,
+  builds them with kustomize and then applies result on cluster.
+  It creates all neccessary kjob CRDs.
+
+  Args:
+    args - user provided arguments
+  Returns:
+    None
+  """
+  temp_dir = tempfile.mkdtemp()
+  mkdir(join(temp_dir, "bases"))
```

Comment:
Can we extract the bases path to a new variable and call these variables something like "kjob_kustomize_path" and "kjob_kustomize_bases_path"? 

It would be then obvious that we are creating a directory within a directory.
Using tmp in name obsures what it really is hurting readability.

Reply:
Resolving as I see it's done.